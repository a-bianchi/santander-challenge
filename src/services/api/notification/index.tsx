//import * as Types from './notification.types';

export class NotificationApi {}
