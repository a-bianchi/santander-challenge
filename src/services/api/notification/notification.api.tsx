import * as Types from './notification.types';

export class NotificationApi {
    private notifications = [
        {
            id: 1,
            userId: '2',
            message: 'Invitación a meetups el dia 20/04/2021',
            date: '10/04/2021',
        },
        {
            id: 2,
            userId: '2',
            message: 'Invitación cancelada a meetups el dia 20/04/2021',
            date: '12/04/2021',
        },
        {
            id: 3,
            userId: '1',
            message: 'Invitación creada con exito para el dia 04/05/2021',
            date: '25/04/2021',
        },
    ];

    /**
     * Get notifications.
     */
    async getUserNotifications(userId: string): Promise<Types.GetNotificationResult> {
        // make the api call

        const notifications = this.notifications.filter((notification) => notification.userId === userId);

        const response = notifications
            ? {
                  ok: true,
                  problem: null,
                  originalError: null,
                  data: notifications,
                  status: 200,
              }
            : {
                  ok: false,
                  problem: 'CLIENT_ERROR',
                  originalError: {
                      name: 'error',
                      code: '401',
                      config: { url: 'null' },
                      isAxiosError: true,
                      toJSON: {},
                  },
                  data: [
                      {
                          id: 0,
                          userId: '',
                          message: '',
                          date: '',
                      },
                  ],
                  status: 401,
              };

        // the typical ways to die when calling an api
        if (!response.ok) {
            return {
                kind: 'unauthorized',
            };
        }

        // eslint-disable-next-line @typescript-eslint/no-explicit-any

        // transform the data into the format we are expecting
        try {
            return { kind: 'ok', notifications: response.data };
        } catch {
            return { kind: 'bad-data' };
        }
    }
}
