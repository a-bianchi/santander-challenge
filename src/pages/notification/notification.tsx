import { ReactElement, useEffect, useState } from 'react';
import { observer } from 'mobx-react-lite';
import { NavBar, NotificationTable } from '../../components';
import { Container } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { useStores } from '../../models';
import { NotificationApi } from '../../services/api/api';

const mockData = [
    {
        message: 'Meetup mock',
        date: '20/02/2021',
    },
];

export const Notification = observer(
    (): ReactElement => {
        const { t } = useTranslation();
        const [notifications, setNotifications] = useState(mockData);
        const notification = new NotificationApi();
        const { userStore } = useStores();

        useEffect(() => {
            const userId = userStore.id || '2';
            const notificationApi = async (): Promise<void> => {
                try {
                    const data = await notification.getUserNotifications(userId);
                    if (data && data.kind === 'ok') {
                        setNotifications(data.notifications);
                    }
                } catch (error) {
                    console.log(`Error: getNotification ${error}`);
                }
            };
            notificationApi();
        }, []);

        return (
            <div>
                <NavBar data-test="NavBar" admin={false} />
                <Container maxWidth="md">
                    <NotificationTable
                        data-test="DataTableNotification"
                        title={t('notification.tableTitle')}
                        data={notifications}
                    />
                </Container>
            </div>
        );
    },
);
