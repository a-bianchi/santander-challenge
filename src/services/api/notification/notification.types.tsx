import { GeneralApiProblem } from '../api.problem';

export interface notification {
    id: number;
    userId: string;
    message: string;
    date: string;
}

export type GetNotificationResult = { kind: 'ok'; notifications: notification[] } | GeneralApiProblem;
