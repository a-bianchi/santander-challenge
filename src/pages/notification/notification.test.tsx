/* global describe expect it */
import { shallow } from 'enzyme';
import { Notification } from './notification';

describe('Notification Page', () => {
    it('should be defined', () => {
        expect(Notification).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = shallow(<Notification />);
        expect(tree.find('.NavBar')).toBeDefined();
        expect(tree.find('.DataTable')).toBeDefined();
        expect(tree).toMatchSnapshot();
    });
});
