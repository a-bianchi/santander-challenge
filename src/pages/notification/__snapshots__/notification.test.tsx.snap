// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Page should render correctly 1`] = `ShallowWrapper {}`;
